### PR TITLE
Support nested models for android save and delete

### PR DESCRIPTION
**Pull request by Amplifiyer**
Support nested models for android save and delete

### CODE REVIEW

**Review by fjnoyp — APPROVED** [review]

File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt [hunk 1]
```diff
@@ -273,4 +272,18 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
         ))
         handler.post { flutterResult.error("AmplifyException", msg, errorMap) }
     }
+
+    @VisibleForTesting
+    fun deserializeNestedModels(serializedModelData: Map<String, Any>): Map<String, Any> {
+        return serializedModelData.mapValues {
+            if(it.value is Map<*, *>) {
```
Comment (Ashish-Nanda) [inline packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt:279]: Should this not check that the field is actually a `model` based on the schema?

File: packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt [hunk 2]
```diff
@@ -311,6 +311,32 @@ class AmplifyDataStorePluginTest {
         )
     }
 
+    @Test
+    fun test_nested_model_serialization() {
```
Comment (Ashish-Nanda) [inline packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt:315]: Is this not testing de-serialization?

**Review by Ashish-Nanda — COMMENTED** [review]
Looks good, some minor comments.

**Review by Ashish-Nanda — APPROVED** [review]
I'll just merge this now and then make any changes that are required.

File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt (hunk 1 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt:279]: I don't believe we support Maps as a field type in `Model` so this is pretty safe assumption to make.

File: packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt (hunk 2 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePluginTest.kt:315]: you are right, it is deserialization.

**Pull request merged**